types/src/shared/utils/structured_data.ts

```diff
@@ -27,3 +31,51 @@ export function getSanitizedHeaders(rawHeaders: string[]) {
     return acc;
   }, []);
 }
+
+export async function guessDelimiter(csv: string): Promise<string | undefined> {
```

Comment:
I know it predates your work but isn't there a perfect library to do that vs us implementing it? cc @flvndvd @fontanierh

Reply:
Not that I know of. Personally prefer a simple approach that we control VS an npm installed blackbox (guessing delimiters has never been an issue in the current system so far)

Reply:
I'm a bit antsy with all the manual CSV handling we do vs using libraries that have been around for many decades and could possibly achieve the same. No strong feeling.

Reply:
We're using `csv-parse` here which is possibly the most popular js library for CSVs